### DISCUSSION TITLE
repo cleanup

### DISCUSSION THREAD

**Discussion by bwbohl** [music-encoding/music-encoding#873] [Ideas]
I stubbed over some files in the repo I was not aware of or do not know what their purpose is. Moreover I think some files should be moved for a cleaner repo structure. The idea of this discussion is to get things clear about those things before issuing a pull request.

- [x] 1. `source/css/.qicon` don't know why this is there or if it is needed I suppose it's there by acident?
- [x] 2. `source/unusedImages.txt` and `source/usedImages.txt` if I gather correctly these are some remnants of the transformation process from markdown guidelines back to TEI-based guidelines. Are theses files still of use or needed?
- [x] 3. `source/tei_odds.rng` is the schema file referenced from the customizations files. (Sidenote: maybe it would be wise to at some point make a transgression to the latest canon TEI version) as it not really part of the codebase actually we should get rid of it; if we want to keep it for convenience purposes, I think it should be moved to `source/validation`
- [ ] 4. `source/validation/mei_odds.odd` and `source/validation/mei_odds.rng` is this really part of our codebase? I don't really think so; shouldn't we move it to a separate repo?
- [ ] 5. `.keep` files in tests. Can anyone explain their purpose?
- [ ] 6. `utils/schematron` as this is definitely third party code, maybe we could integrate it as submodule? Apart from that: where and when is it needed?

Whether you can answer my questions, have some similar of your own, or just want to give your five cents, please reply and contribute ;-)

**Comment by rettinghaus** [comment]
- [x] The local `tei_odds.rng` is totally outdated and not used anywhere, every other file references the latest version url. (opened a PR directly in #876)
- [x] `.qicon` is something coming from QNAP's Qsync (many people complain about).
- [ ] No file from `utils/schematron` is used anywhere (as far as I can see), so I'm in favor of removing it.  The corresponding repo is https://github.com/Schematron/stf

**Comment by rettinghaus** [comment]
@ahankinson @kepper Do you have any idea about utils/schematron? Is this needed in the repo at all?

**Comment by ahankinson** [comment]
Previously, there was an attempt at a) building a test set for MEI and b) validating both the schema and the schematron rules for every commit. 

I still think there is a need for this, but most of the other stuff has been stripped out. So I guess it can be scrapped?

**Comment by kepper** [comment]
The stuff in that folder is rather old, yes. More recent work in that area (though slightly different) can be found in [this file](https://github.com/music-encoding/music-encoding/blob/a9c54002b375619740e0e4a01538d568892abbf2/source/validation/mei-source.sch), discussed in [this issue](https://github.com/music-encoding/music-encoding/issues/790). Before scrapping it, I'd like to check if things can be moved there. Not much time for the next ten days, though…

**Comment by ahankinson** [comment]
- `.keep` files are necessary to retain a folder in git, since it operates on a file level, and only preserves folders if there is a file in them. They were used so that we would have a folder for each customization where you could put test files in them, and they would automatically run a test suite against them. That plan didn't make it all the way.
 - The `mei_odds` files are local copies of the ODD for TEI. [They're still referenced in the customization files](https://github.com/music-encoding/music-encoding/blob/develop/customizations/mei-CMN.xml#L20-L21).
 - `utils/schematron` was used when we needed to build the Schematron rules to validate the changes on commit. Maybe this has changed now?
